pkg/istio-agent/sds-agent.go

```diff
@@ -173,26 +162,20 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig,
 	if citadel.ProvCert != "" {
 		certDir = citadel.ProvCert
 	}
-	if _, err := os.Stat(certDir + "/key.pem"); err == nil {
-		sa.CertsPath = certDir
-	}
-	if sa.CertsPath != "" {
-		log.Warna("Using existing certificate ", sa.CertsPath)
-	}
-
 	// If the root-cert is in the old location, use it.
 	if _, err := os.Stat(certDir + "/root-cert.pem"); err == nil {
+		log.Warna("Using existing certificate ", certDir)
 		CitadelCACertPath = certDir
 	}
 
 	if sa.secOpts.CAEndpoint == "" {
 		// if not set, we will fallback to the discovery address
 		sa.secOpts.CAEndpoint = discAddr
 	}
-
 	// Next to the envoy config, writeable dir (mounted as mem)
-	sa.secOpts.WorkloadUDSPath = LocalSDS
-	sa.secOpts.CertsDir = sa.CertsPath
+	if sa.secOpts.WorkloadUDSPath == "" {
```

Comment:
Is this introducing a different behavior?

Reply:
Actually, i think it is a bug ignoring `sa.secOpts.WorkloadUDSPath`